extensions/applicationinsights-clickanalytics-js/src/Interfaces/Datamodel.ts

```diff
@@ -0,0 +1,382 @@
+    /**
+     * Get element content 
+     */
+    getElementContent: (element: Element, eventType?: EventType) => IContent;
+    
+  }
+
+/**
+ * Page Action event
+ */
+export interface IPageActionTelemetry extends ITelemetryEventInternal {
+    /**
+     * Target uri for PageAction events
+     */
+    targetUri?: string;
+    /**
+     *  One of the awa.actionType values
+     */
+    actionType?: string;
+    /**
+     * One of the awa.behavior values. 
+     */
+    behavior?: number;
+    /**
+     * X, Y representing the absolute co-ordinates withrespect to topleft corner of the page. This should be normalized for the screen resolution to provide better heat map.
+     */
+    clickCoordinates?: string;
+    /**
+     * JSON-formatted array of content acted upon 
+     */
+    content?: any;
+    /**
+     * Version indicating content version which aids in parsing the content.
+     */
+    contentVer?: string;
+    /**
+     * Uri of the referrer, this is a convinence for adaptors to just leverage PageAction for click analytics without linking it to the respective Page Views.
+     */
+    refUri?: string;
+    /**
+     * Time taken in milliseconds since the user saw the page and took the action (such as click on a link, etc.). This will be in seconds.
+     */
+    timeToAction?: number;
+}
+
+export interface ITelemetryEventProperties {
+    /**
+     * User specified custom properties
+     */
+    [name: string]: string | number | boolean | string[] | number[] | boolean[] | object;
+}
+
+/**
+ * Page Action event properties (part C)
+ */
+export interface IPageActionProperties extends ITelemetryEventProperties {
+    /**
+     * Uri of the referrer, this is a convinence for adaptors to just leverage PageAction for click analytics without linking it to the respective Page Views.
+     */
+    refUri?: string;
+    /**
+     * Time taken in milliseconds since the user saw the page and took the action (such as click on a link, etc.). This will be in seconds.
+     */
+    timeToAction?: number;
+    /**
+     * Server impression Guid, as set by the adopter
+     */
+    serverImpressionGuid?: string;
+    /**
+     * List of cookies in semi-colon delimited name value pair
+     */
+    cookies?: string;
+}
+
+export interface ITelemetryEvent {
+    /**
+     * Page name.
+     */
+    name?: string;
+    /**
+     * A relative or absolute URL that identifies the page or other item. Defaults to the window location.
+     */
+    uri?: string;
+    /**
+     * Represents locale of the Site user has selected
+     */
+    market?: string;
+    /**
+     * Page type
+     */
+    pageType?: string;
+    /**
+     * boolean is user logged in
+     */
+    isLoggedIn?: boolean;
+    /**
+     * property bag to contain an extension to domain properties - extension to Part B
+     */
+    properties?: {
+        [key: string]: any;
+    };
+}
+
+export interface ITelemetryEventInternal extends ITelemetryEvent {
```

Comment:
This interface was used to differentiate 1DS and ApplicationInsights, most likely this could be removed entirely